src/wallet/components/WalletDropdownFundLink.tsx

```diff
@@ -14,69 +12,38 @@ export function WalletDropdownFundLink({
   rel,
   target,
   text = 'Fund wallet',
+  fundingUrl,
 }: WalletDropdownFundLinkReact) {
-  const [fundingUrl, setFundingUrl] = useState('');
+  const isCoinbaseSmartWallet = useIsWalletACoinbaseSmartWallet();
 
-  const iconSvg = useIcon({ icon });
-
-  useEffect(() => {
-    const currentURL = window.location.href;
-    const tabName = document.title;
-    const url = `http://keys.coinbase.com/fund?dappName=${encodeURIComponent(
-      tabName,
-    )}&dappUrl=${encodeURIComponent(currentURL)}&version=${encodeURIComponent(
-      version,
-    )}&source=onchainkit`;
-    setFundingUrl(url);
-  }, []);
-
-  const handleClick = useCallback(
-    (e: React.MouseEvent) => {
-      e.preventDefault();
-      const { width, height } = getWindowDimensions(popupSize);
-
-      const left = Math.round((window.screen.width - width) / 2);
-      const top = Math.round((window.screen.height - height) / 2);
-
-      const windowFeatures = `width=${width},height=${height},resizable,scrollbars=yes,status=1,left=${left},top=${top}`;
-      window.open(fundingUrl, target, windowFeatures);
-    },
-    [fundingUrl, popupSize, target],
-  );
-
-  const overrideClassName = cn(
-    pressable.default,
-    'relative flex items-center px-4 py-3 w-full',
-    className,
-  );
-
-  const linkContent = useMemo(
-    () => (
-      <>
-        <div className="-translate-y-1/2 absolute top-1/2 left-4 flex h-[1.125rem] w-[1.125rem] items-center justify-center">
-          {iconSvg}
-        </div>
-        <span className={cn(themeText.body, 'pl-6')}>{text}</span>
-      </>
-    ),
-    [iconSvg, text],
-  );
+  if (fundingUrl) {
```

Comment:
I'm not sure I like this design approach with the conditional rendering of 3 different components (and prop destructuring)

I think we should keep this file, and switch the `fundingUrl` and `handleClick` behavior based on the presence of `walletCapabilities`

cc: @alessey @alissacrane-cb what do you guys think?

Reply:
I'm no FE expert, but the React docs say that using an effect to update state or cache expensive calculations is an anti-pattern (https://react.dev/learn/you-might-not-need-an-effect). That's why I wrapped getting the URLs in useMemo calls.

The conditional rendering is really for maintainability reasons. The fund link component was already getting kinda complex, I didn't want to add even more logic to it. You can imagine that if we add more funding integrations that component could get really out of hand.

I feel like separating the logic of getting the right URL from the logic of actually rendering the button that handles dealing with the popup / new tab opening is a clear separation of concern, and makes it way easier to keep our unit test coverage at 100%.

Reply:
Oh and I can get rid of the props spread if you want, I was mainly doing it because those props should be a pass through for the wallet specific components. And those components are not exported, and the props are all strongly typed, so I'm not really sure why it would be an anti-pattern here. But I'm happy to make it all verbose if that's what we want. We should probably turn on the linting rule for no prop spreading if that's the case.

Reply:
@alessey @alissacrane-cb would love your thoughts on this architecture!

Reply:
I agree with the goal of extracting the funding url logic, but I think you could reduce boiler plate and keep your separation of concerns having a useFundingUrl hook which called the appropriate util.

Reply:
@alessey yeah I like that idea, I've refactored this.